File: tournaments/integerToStringOfFixedWidth/integerToStringOfFixedWidth.cs
string integerToStringOfFixedWidth(int number, int width) {
    var result = new StringBuilder();
    var count = 0;

    while (0 < number && width > count) {
        result.Insert(0, (number % 10).ToString());
        number /= 10;
        ++count;
    }

    return result.ToString().PadLeft(width, '0');
}

File: tournaments/integerToStringOfFixedWidth/integerToStringOfFixedWidth.cpp
std::string integerToStringOfFixedWidth(int number, int width) {

  std::string result = "";

  for (int i = 0; i < width; i++) {
    result += '0';
  }

  int position = width - 1;
  while (number && position >= 0) {
    result[position] = '0' + number % 10;
    number /= 10;
    position--;
  }

  return result;
}
